src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -776,6 +760,11 @@ EnsureRelationCanBeDistributed(Oid relationId, Var *distributionColumn,
 	ErrorIfUnsupportedConstraint(relation, distributionMethod, distributionColumn,
 								 colocationId);
 
+	if (TableReferenced(relationId) || TableReferencing(relationId))
+	{
+		InvalidateForeignKeyGraph();
```

Comment:
Is this in the right place?  (EnsureRelationCanBeDistributed)  almost looks like a merge conflict

Reply:
The rationale was that right after we've passed `ErrorIfUnsupportedConstraint ()`, we're sure that the table is going to be distributed. And, the table is locally created. So, if the foreign keys exist with the table, we could send the invalidations.

However, I see your point. I'll move it to right after we create the shards & add the metadata.